wp-content/plugins/core/src/Blocks/Types/Media_Text/Media_Text.php

```diff
@@ -3,77 +3,150 @@
-			->add_data_source( 'className-c-block', self::LAYOUT )
-			->add_data_source( 'className-c-block', self::WIDTH )
-			->set_dashicon( 'menu-alt' )
-			->add_content_section( $this->child_container() )
-			->add_toolbar_section( $this->layout_toolbar() )
-			->build();
+	/**
+	 * Register the block
+	 */
+	public function add_block() {
+		$this->set_block( new Block( self::NAME, [
+			'title'       => __( 'Media + Text', 'tribe' ),
+			'description' => __( 'An image or video plus text content with several layout options.', 'tribe' ),
+			'icon'        => '<svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" focusable="false"><path d="M13 17h8v-2h-8v2zM3 19h8V5H3v14zM13 9h8V7h-8v2zm0 4h8v-2h-8v2z"></path></svg>',
+			'keywords'    => [ __( 'image', 'tribe' ), __( 'video', 'tribe' ), __( 'display', 'tribe' ), __( 'text', 'tribe' ) ],
+			'category'    => 'layout',
+			'supports'    => [ 'align' => false ],
+		] ) );
 	}
 
-	private function child_container(): Content_Section {
-		return $this->factory->content()->section()
-			->add_class( 'b-media-text__container' )
-			->add_field(
-				$this->factory->content()->field()->fixed_container( self::CONTAINER )
-					->add_template_block( Media_Text_Media::NAME )
-					->add_template_block( Media_Text_Text::NAME )
-					->merge_nested_attributes( Media_Text_Media::NAME )
-					->merge_nested_attributes( Media_Text_Text::NAME )
-					->build()
-			)
-			->build();
+	/**
+	 * Register Fields for block
+	 */
+	public function add_fields() {
+		$this->add_field( new Field( self::NAME . '_' . self::TITLE, [
+				'label' => __( 'Title', 'tribe' ),
+				'name'  => self::TITLE,
+				'type'  => 'text',
+			] )
+		)->add_field( new Field( self::NAME . '_' . self::CONTENT, [
+				'label'        => __( 'Description', 'tribe' ),
+				'name'         => self::CONTENT,
+				'type'         => 'wysiwyg',
+				'toolbar'      => 'basic',
+				'media_upload' => 0,
+			] )
+		)->add_field( new Field( self::NAME . '_' . self::CTA, [
+				'label' => __( 'Call to Action', 'tribe' ),
+				'name'  => self::CTA,
+				'type'  => 'link',
+			] )
+		)->add_field( new Field( self::NAME . '_' . self::MEDIA_TYPE, [
+				'label'   => __( 'Media Type', 'tribe' ),
+				'name'    => self::MEDIA_TYPE,
+				'type'    => 'radio',
+				'choices' => [
+					self::IMAGE => __( 'Image', 'tribe' ),
+					self::EMBED => __( 'Video oEmbed', 'tribe' ),
+				],
+				'default_value'   => [
+					self::IMAGE,
+				],
+			] )
+		)->add_field( new Field( self::NAME . '_' . self::IMAGE, [
+				'label'             => __( 'Image', 'tribe' ),
+				'name'              => self::IMAGE,
+				'type'              => 'image',
+				'preview_size'      => 'medium',
+				'instructions'      => __( 'Recommended image size by layout:<br>Center: 1920px wide with a 16:9 aspect ratio.<br>Left/Right: 1536px wide with a 4:3 aspect ratio.', 'tribe' ),
+				'conditional_logic' => [
```

Comment:
This doesn't appear to work. Thoughts? Same with the Video field below.

Reply:
Fields get prefixed with their type in the ACF_Configuration class so you need to prepend `field_`
`'field_' . self::NAME . '_' . self::MEDIA_TYPE,`
`'field_' . self::NAME . '_' . self::MEDIA_TYPE`

Reply:
Purrfect. thank you!